src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1293,6 +1293,18 @@ torch::Tensor VideoDecoder::convertFrameToTensorUsingFilterGraph(
   return tensor;
 }
 
+VideoDecoder::~VideoDecoder() {
+  for (auto& [streamIndex, stream] : streams_) {
+    auto& device = stream.options.device;
+    if (device.type() == torch::kCPU) {
+    } else if (device.type() == torch::kCUDA) {
+      releaseContextOnCuda(device, stream.codecContext.get());
+    } else {
+      TORCH_CHECK(false, "Invalid device type: " + device.str());
```

Comment:
Just to make sure I understand, this will never happen because it was checked before during creation, right?

Reply:
Yes, it should crash before this check